### ISSUE TITLE
the 0 results of val.py

### ISSUE THREAD

**Issue by G-Apple1** [ultralytics/yolov5#4344]
## ❔Question

I made some changes to your network model (add SE module to the Conv ), and the trained model achieved a little improvement on my test set ( I saw the final val result at the end of training ).
But when i run val.py with the --weights run/train/exp/weights/last.pt, the result P, R, map@0.5 and so on all is 0 .

**The results of train**

Epoch   gpu_mem       box       obj       cls    labels  img_size
   199/199     3.47G   0.01118   0.03534  0.001137        23       960: 100%|██████████| 492/492 [05:17<00:00,  1.55it/s]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|██████████| 115/115 [00:24<00:00,  4.73it/s]
                 all        460       5189      0.827      0.827      0.875      0.691
                   0        460       1872      0.792      0.759      0.825      0.641
                   1        460       1611      0.878      0.757      0.851      0.626
                   2        460       1706       0.81      0.966      0.948      0.806
200 epochs completed in 19.047 hours.
Optimizer stripped from runs/train/exp6/weights/last.pt, 93.8MB
Optimizer stripped from runs/train/exp6/weights/best.pt, 93.8MB
Results saved to runs/train/exp6

**The results of val**
(yolov5) scau2@scau2-desktop:/media/scau2/1T2/xcg/YOLOV5/yolov5$ python val.py --data p_datasets.yaml --img 960 --weights runs/train/exp6/weights/best.pt --device  0 --task val --half
val: data=./data/p_datasets.yaml, weights=['runs/train/yolov5s-se300/weights/best.pt'], batch_size=8, imgsz=960, conf_thres=0.001, iou_thres=0.6, task=val, device=0, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=exp, exist_ok=--exist-ok, half=True
YOLOv5 🚀 v5.0-339-g53bfcbe torch 1.7.0 CUDA:0 (GeForce GTX 980 Ti, 6080.8125MB)

Fusing layers... 
Model Summary: 637 layers, 7364072 parameters, 0 gradients, 16.3 GFLOPs
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|█| 58/58
                 all        460          0          0          0          0          0:  97%|▉| 56/58
Speed: 0.4ms pre-process, 7.3ms inference, 0.2ms NMS per image at shape (8, 3, 960, 960)
val: Scanning '../p_datasets/labels/test.cache' images and labels... 460 found, 0 missing, 0 empty, 0
Results saved to runs/val/exp

## Additional context
i run detect.py with the --weights run/train/exp/weights/last.pt, the predictions is poor.

## Add SE module to the Conv as follow
class SELayer(nn.Module):
    def __init__(self, channel, reduction=16):
        super(SELayer, self).__init__()
        self.avg_pool = nn.AdaptiveAvgPool2d(1)
        self.fc = nn.Sequential(
            nn.Linear(channel, channel // reduction, bias=False),
            nn.ReLU(inplace=True),
            nn.Linear(channel // reduction, channel, bias=False),
            nn.Sigmoid()
        )

    def forward(self, x):
        b, c, _, _ = x.size()
        y = self.avg_pool(x).view(b, c)
        y = self.fc(y).view(b, c, 1, 1)
        return x * y.expand_as(x)

class Conv(nn.Module):
    # Standard convolution
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):  # ch_in, ch_out, kernel, stride, padding, groups
        super().__init__()
        self.conv = nn.Conv2d(c1, c2, k, s, autopad(k, p), groups=g, bias=False)
        self.bn = nn.BatchNorm2d(c2)
        self.act = nn.SiLU() if act is True else (act if isinstance(act, nn.Module) else nn.Identity())
        self.se_conv = SE_Conv(c2)

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def forward_fuse(self, x):
        return self.se_conv(self.act(self.conv(x)))

**Comment by G-Apple1** [comment]
**I added the SE module to the conv module of common.py**, and the number of parameters increased. I don’t know if the addition is correct, but there was no error during training, and the evaluation indicators were all normal, 

but after training. When runs/train/exp/weights/best.pt is used to run val.py, the evaluation indicators are all 0.

I found that I just need to change class Conv `def forward_fuse(self, x): return self.se_conv(self.act(self.conv(x))) `here to `return self.act(self.conv(x)) `, and run val .py can reproduce the training indicators.

I suspect that the SE module I added has not been trained, or the model loaded by train.py is different from the model loaded by val.py

**Comment by glenn-jocher** [comment]
@G-Apple1 if you modify modules you need to make those same modules available anytime the model is loaded, otherwise there will be problems. i.e. you can't train custom layers in one environment and evaluate in a master clone.

**Comment by G-Apple1** [comment]
> @G-Apple1 if you modify modules you need to make those same modules available anytime the model is loaded, otherwise there will be problems. i.e. you can't train custom layers in one environment and evaluate in a master clone.

Thank you for your reminding. I have solved this problem. The reasons for this problem are as follows: 
The Conv module of the train.py model calls the forward function, while the Conv module of the val.py model calls the fuseforward function
